From 06302e8ebcc188f247071b089dac1ee45cae614e Mon Sep 17 00:00:00 2001
From: Laurentiu Palcu <laurentiu.palcu@nxp.com>
Date: Mon, 2 Jul 2018 13:03:38 +0300
Subject: [PATCH 4127/5242] MLK-18728: drm: imx: hdp: fix 4K@60 on mscale

commit  f3b1b2d2d6bab76a55fd7dae016d4fcbdc602f68 from
https://source.codeaurora.org/external/imx/linux-imx.git

On mscale, 4K@60 uses YUV420. However, the following commit:

8eeed3553bd1: MLK-18690-1: hdmi: Rewrite hdmi avi infoframe function

moved to using the kernel built-in infoframe creation routine, instead
of Cadence's one, but skipped handling the various colorspace formats.

Signed-off-by: Laurentiu Palcu <laurentiu.palcu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/drm/imx/hdp/imx-hdmi.c |   15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git a/drivers/gpu/drm/imx/hdp/imx-hdmi.c b/drivers/gpu/drm/imx/hdp/imx-hdmi.c
index 84fa5be..5462a2e 100644
--- a/drivers/gpu/drm/imx/hdp/imx-hdmi.c
+++ b/drivers/gpu/drm/imx/hdp/imx-hdmi.c
@@ -127,6 +127,21 @@ static int hdmi_avi_info_set(struct imx_hdp *hdp,
 					  HDMI_COLORIMETRY_NONE;
 	frame.extended_colorimetry = ext_colorimetry;
 
+	switch (format) {
+	case YCBCR_4_4_4:
+		frame.colorspace = HDMI_COLORSPACE_YUV444;
+		break;
+	case YCBCR_4_2_2:
+		frame.colorspace = HDMI_COLORSPACE_YUV422;
+		break;
+	case YCBCR_4_2_0:
+		frame.colorspace = HDMI_COLORSPACE_YUV420;
+		break;
+	default:
+		frame.colorspace = HDMI_COLORSPACE_RGB;
+		break;
+	}
+
 	ret = hdmi_avi_infoframe_pack(&frame, buf + 1, sizeof(buf) - 1);
 	if (ret < 0) {
 		DRM_ERROR("failed to pack AVI infoframe: %d\n", ret);
-- 
1.7.9.5

